src/http/server/response.cr

```diff
@@ -40,6 +40,26 @@ class HTTP::Server
       output.response = self
     end
 
+    def remote_address
+      case @io
```

Comment:
Make this `case io = @io`. Using the local variable `io` in each branch will have it restricted to the appropriate type.